src/autodiff.jl

```diff
@@ -382,6 +382,136 @@ function residual_jacobian!(arrays::Jacobian,
     return nothing
 end
 
+function residual_jacobian_adj!(arrays::Jacobian,
+                             residual_adj_polar!,
+                             v_m, v_a, ybus_re, ybus_im, pinj, qinj, pv, pq, ref, nbus,
+                             type::AbstractJacobian)
+    nvbus = length(v_m)
+    ninj = length(pinj)
+    F = zeros(Float64, length(arrays.t1sF))
+    adj_F = zeros(Float64, length(arrays.t1sF))
+    if isa(type, StateJacobian)
+        arrays.x[1:nvbus] .= v_m
+        arrays.x[nvbus+1:2*nvbus] .= v_a
+        global dJ = zeros(Float64, length(F), length(arrays.map))
```

Comment:
Why use `global` here?

Reply:
I assume that's because of the scope of the variables defined in the `if` statement. I think it would be more appropriate to define `dJ` before line 393, with
```
local dJ
```

Reply:
This is gone.